From 9f27d6f51c34d96179c6aafc0f33266d8ac13f9d Mon Sep 17 00:00:00 2001
From: Ryne Everett <ryneeverett@gmail.com>
Date: Sat, 15 Aug 2026 23:14:49 -0400
Subject: [PATCH 128/137] Assembly: fix errors using Fixed Joint on a Part

Resolve #31622.

This addresses the minimum typing errors to make Fixed Joints work on
Parts and so to avoid AttributeError. There are a couple other instances
in this module where the `None` possibly returned from `getAssembly` is
not handled. I can address these here if preferred.
---
 src/Mod/Assembly/JointObject.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/Mod/Assembly/JointObject.py b/src/Mod/Assembly/JointObject.py
index a29299b82..96e275087 100644
--- a/src/Mod/Assembly/JointObject.py
+++ b/src/Mod/Assembly/JointObject.py
@@ -838,7 +838,7 @@ class Joint:
     def setJointConnectors(self, joint, refs):
         # current selection is a vector of strings like "Assembly.Assembly1.Assembly2.Body.Pad.Edge16" including both what selection return as obj_name and obj_sub
         assembly = self.getAssembly(joint)
-        isAssembly = assembly.Type == "Assembly"
+        isAssembly = assembly is not None and assembly.Type == "Assembly"
 
         if len(refs) >= 1:
             joint.Reference1 = refs[0]
@@ -928,7 +928,7 @@ class Joint:
         if not part1 or not part2:
             return False
 
-        isAssembly = assembly.Type == "Assembly"
+        isAssembly = assembly is not None and assembly.Type == "Assembly"
         if isAssembly:
             joint.Suppressed = True
             part1Connected = assembly.isPartConnected(part1)
@@ -1228,7 +1228,7 @@ class ViewProviderJoint:
         # Assuming Reference1 corresponds to the first part link
         if hasattr(self.app_obj, "Reference1"):
             part = UtilsAssembly.getMovingPart(self.app_obj.Reference1)
-            if part is not None and not assembly.isPartConnected(part):
+            if part is not None and assembly is not None and not assembly.isPartConnected(part):
                 overlays[Gui.IconPosition.BottomLeft] = "Part_Detached"
 
         return overlays
-- 
2.43.0

